src/integrationTest/java/uk/gov/hmcts/darts/testutils/IntegrationBase.java

```diff
@@ -55,19 +125,61 @@ static void configureProperties(DynamicPropertyRegistry registry) {
 
     @BeforeEach
     void clearDb() {
+        resetSequences();
         dartsDatabase.clearDatabaseInThisOrder();
+        resetTablesWithPredefinedTestData();
     }
 
     @AfterEach
     void clearTestData() {
         logAppender.reset();
     }
 
+    public void resetSequences() {
+        try (EntityManager em = entityManagerFactory.createEntityManager()) {
+            em.getTransaction().begin();
+            final Query query = em.createNativeQuery("SELECT sequence_name FROM information_schema.sequences WHERE sequence_schema = 'darts'");
+            final List sequences = query.getResultList();
+            for (Object seqName : sequences) {
+                if (SEQUENCES_RESET_FROM.contains(seqName.toString())) {
+                    em.createNativeQuery("ALTER SEQUENCE darts." + seqName + " RESTART WITH " + SEQUENCE_START_VALUE).executeUpdate();
+                } else if (!SEQUENCES_NO_RESET.contains(seqName.toString())) {
+                    em.createNativeQuery("ALTER SEQUENCE darts." + seqName + " RESTART").executeUpdate();
+                }
+            }
+            em.getTransaction().commit();
+        }
+    }
+
+    @Transactional
+    public void resetTablesWithPredefinedTestData() {
+
+        retentionPolicyTypeRepository.deleteAll(
+            retentionPolicyTypeRepository.findByIdGreaterThanEqual(SEQUENCE_START_VALUE)
+        );
+
+        eventHandlerRepository.deleteAll(
+            eventHandlerRepository.findByIdGreaterThanEqual(SEQUENCE_START_VALUE)
+        );
+
+        automatedTaskRepository.deleteAll(
+            automatedTaskRepository.findByIdGreaterThanEqual(SEQUENCE_START_VALUE)
+        );
+
+        userAccountRepository.deleteAll(
+            userAccountRepository.findByIdGreaterThanEqual(SEQUENCE_START_VALUE)
+        );
+
+        securityGroupRepository.deleteAll(
+            securityGroupRepository.findByIdGreaterThanEqual(SEQUENCE_START_VALUE)
+        );
+    }
+
```

Comment:
Consider relocating this code to `DartsDatabaseStub` which handles management of database data.

Reply:
nice one, will do cheers